Terminal.Gui/View/Layout/Pos.cs

```diff
@@ -0,0 +1,694 @@
+    ///     <see cref="View"/>.
+    /// </summary>
+    /// <returns>The <see cref="Pos"/> that depends on the other view.</returns>
+    /// <param name="view">The <see cref="View"/>  that will be tracked.</param>
+    public static Pos Right (View view) { return new PosView (view, Side.Right); }
+
+    /// <summary>
+    ///     Gets a position that is anchored to a certain point in the layout. This method is typically used
+    ///     internally by the layout system to determine where a View should be positioned.
+    /// </summary>
+    /// <param name="width">The width of the area where the View is being positioned (Superview.ContentSize).</param>
+    /// <returns>
+    ///     An integer representing the calculated position. The way this position is calculated depends on the specific
+    ///     subclass of Pos that is used. For example, PosAbsolute returns a fixed position, PosAnchorEnd returns a
+    ///     position that is anchored to the end of the layout, and so on.
+    /// </returns>
+    internal virtual int Anchor (int width) { return 0; }
+
+    /// <summary>
+    ///     Calculates and returns the position of a <see cref="View"/> object. It takes into account the dimension of the
+    ///     superview and the dimension of the view itself.
+    /// </summary>
+    /// <param name="superviewDimension">
+    ///     The dimension of the superview. This could be the width for x-coordinate calculation or the
+    ///     height for y-coordinate calculation.
+    /// </param>
+    /// <param name="dim">The dimension of the View. It could be the current width or height.</param>
+    /// <param name="us">The View that holds this Pos object.</param>
+    /// <param name="dimension">Width or Height</param>
+    /// <returns>
+    ///     The calculated position of the View. The way this position is calculated depends on the specific subclass of Pos
+    ///     that
+    ///     is used.
+    /// </returns>
+    internal virtual int Calculate (int superviewDimension, Dim dim, View us, Dimension dimension) { return Anchor (superviewDimension); }
+
+    /// <summary>
+    ///     Diagnostics API to determine if this Pos object references other views.
+    /// </summary>
+    /// <returns></returns>
+    internal virtual bool ReferencesOtherViews () { return false; }
+}
+
+/// <summary>
+///    Represents an absolute position in the layout. This is used to specify a fixed position in the layout.
+/// </summary>
+/// <remarks>
+///     <para>
+///     This is a low-level API that is typically used internally by the layout system. Use the various static
+///     methods on the <see cref="Pos"/> class to create <see cref="Pos"/> objects instead.
+///     </para>
+/// </remarks>
+/// <param name="position"></param>
+public class PosAbsolute (int position) : Pos
+{
+    /// <summary>
+    ///    The position of the <see cref="View"/> in the layout.
+    /// </summary>
+    public int Position { get; } = position;
```

Comment:
What does this even mean?

It's not tested and it's not used elsewhere, except for a really odd use pointed out in the next comment.

Reply:
This is the value for an absolute Pos. 

If a dev does `view.X = 1`, a `PosAbsolute (1)` is created and the `1` is stored in `Position`. This previously was named `n`.

It is extensively tested (just not specifically).

Reply:
If there's not an explicit test on a public API element, it's not tested.

All it takes is to change that line and some things will break while others won't, and the reason is obscured slightly, especially the way these types are used.

But, like in my other comments, this is all related to the same small set of core issues that I'll write once rather than scattered around and hard to follow. :)

Reply:
Added a test. Thanks.

Reply:
Yeah. Annoying, but unit tests gonna unit test, and properties gonna property.